fastestimator/pipeline.py

```diff
@@ -139,102 +139,111 @@ def get_per_epoch(ops: Iterable[Union[T, Scheduler[T]]], epoch: int) -> List[T]:
 
 
 class BasePipeline:
-    def get_modes(self) -> Set[str]:
-        raise NotImplementedError()
-
-    def get_global_batch_size(self, mode: str, epoch: int) -> int:
-        raise NotImplementedError()
-
-    def get_signature_epochs(self, mode: str) -> List[int]:
-        raise NotImplementedError()
-
-    def get_all_output_keys(self) -> Set[str]:
+    def __init__(self,
```

Comment:
basePipeline isn't the right place to specify the init since then anyone extending it for a new pipeline with different data types would have to edit the BasePipeline too